domain/secret/service/service.go

```diff
@@ -682,7 +729,7 @@ func (s *SecretService) ProcessCharmSecretConsumerLabel(
 
 	if uri == nil {
 		var err error
-		uri, err = s.GetURIByConsumerLabel(ctx, label, unitName)
+		uri, err = s.secretState.GetURIByConsumerLabel(ctx, label, unitName)
```

Comment:
This and also UpdateCharmSecret() a few lines above aren't RunAtomic so the entire ProcessCharmSecretConsumerLabel call isn't atomic

Reply:
No, we can't run UpdateCharmSecret in atomic transaction because the UpdateCharmSecret does leadership token check and also updates refCount.

Reply:
We've already done the token check at the start of this method, so the UpdateCharmSecret above which takes the leader token need to be refactored so that we pass in isLeader.
Also, all we're doing here is updating the label, not the content. so no ref count is involved at all. The TODO indicates we need to rework this logic to do the label update when the consumed revision is looked up, accounting for cross model secrets. Let's try and get that done and see how it plays out. Last resort could be to add a UpdateSecretLabel() method to remove ambiguity, but better if it can be done with the consumed revision operation - that already has the update label support
`GetConsumedRevision(ctx context.Context, uri *secrets.URI, unitName string, refresh, peek bool, labelToUpdate *string) (int, error)`